lib/kafo/scenario_manager.rb

```diff
@@ -0,0 +1,123 @@
+# encoding: UTF-8
+require 'kafo_wizards'
+
+module Kafo
+  class ScenarioManager
+    attr_reader :config_dir, :last_scenario_link, :previous_scenario
+
+    def initialize(config, last_scenario_link_name='last_scenario.yaml')
+      @config_dir = File.file?(config) ? File.dirname(config) : config
+      @last_scenario_link = File.join(config_dir, last_scenario_link_name)
+      @previous_scenario = File.realpath(last_scenario_link) if File.exists?(last_scenario_link)
+    end
+
+    def available_scenarios
+      # assume that *.yaml file in config_dir that has key :name is scenario definition
+      @available_scenarios ||= Dir.glob(File.join(config_dir, '*.yaml')).reject { |f| f =~ /#{last_scenario_link}$/ }.inject({}) do |scns, scn_file|
+        begin
+          content = YAML.load_file(scn_file)
+          scns[scn_file] = content if content.has_key?(:name)
```

Comment:
The if here makes name value in config file mandatory, on old answer file this might not be present. At least we should make https://github.com/theforeman/kafo/blob/master/config/kafo.yaml.example#L7 uncommented and document it. Kafofy luckily generates the name for quite long time (since 0.4) so such old configurations might not exist widely.